Napzak-iOS/Napzak-iOS/Network/Interest/InterestService.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  InterestService.swift
+//  Napzak-iOS
+//
+//  Created by 조혜린 on 4/28/25.
+//
+
+import Moya
+
+protocol InterestServiceProtocol {
+    func postInterest(productId: Int, completion: @escaping (NetworkResult<Any>) -> ())
+    func deleteInterest(productId: Int, completion: @escaping (NetworkResult<Any>) -> ())
+}
+
+final class InterestService: BaseService, InterestServiceProtocol {
+    
+    private let provider = MoyaProvider<InterestAPI>.init(plugins: [MoyaPlugin()])
+    
+    func postInterest(productId: Int, completion: @escaping (NetworkResult<Any>) -> ()) {
+        request(provider, .postInterest(productId: productId), completion: completion)
+    }
+    
+    func deleteInterest(productId: Int, completion: @escaping (NetworkResult<Any>) -> ()) {
+        request(provider, .deleteInterest(productId: productId), completion: completion)
+    }
+}
```

Comment:
장르 관련 API 서비스객체와 내용 동일

```suggestion

import Foundation
import Moya
import os

protocol InterestServiceProtocol {
    func postInterest(productId: Int) async -> Result<Void, NetworkError>
    func deleteInterest(productId: Int) async -> Result<Void, NetworkError>
}

final class InterestService: BaseService, InterestServiceProtocol {
    
    private let provider = MoyaProvider<InterestAPI>.init(plugins: [MoyaPlugin()])
    
    
    func postInterest(productId: Int) async -> Result<Void, NetworkError> {
        return await request(provider, .postInterest(productId: productId))
    }
    
    func deleteInterest(productId: Int) async -> Result<Void, NetworkError> {
        return await request(provider, .deleteInterest(productId: productId))
    }
}
```